pkg/daemon/podprobe/prober.go

```diff
@@ -86,10 +107,91 @@ func (pb *prober) runProbe(p *appsv1alpha1.ContainerProbeSpec, probeKey probeKey
 		klog.InfoS("TCP-Probe Host", "host", host, "port", port, "timeout", timeout)
 		return pb.tcp.Probe(host, port, timeout)
 	}
+
 	klog.InfoS("Failed to find probe builder for container", "containerName", containerRuntimeStatus.Metadata.Name)
 	return probe.Unknown, "", fmt.Errorf("missing probe handler for %s", containerRuntimeStatus.Metadata.Name)
 }
 
+func newRequestForHTTPGetAction(httpGet *v1.HTTPGetAction, podIP string, userAgentFragment string) (*http.Request, error) {
+	scheme := string(httpGet.Scheme)
+	if scheme == "" {
+		scheme = "http"
+	}
+
+	host := httpGet.Host
+	if host == "" {
+		host = podIP
+	}
+
+	port := httpGet.Port.IntValue()
+	if port <= 0 || port >= 65536 {
+		return nil, fmt.Errorf("invalid port number: %v", port)
+	}
+	path := httpGet.Path
+	url := formatURL(scheme, host, port, path)
+	headers := v1HeaderToHTTPHeader(httpGet.HTTPHeaders)
+
+	return newProbeRequest(url, headers, userAgentFragment)
+}
+
+// formatURL formats a URL from args.  For testability.
+func formatURL(scheme string, host string, port int, path string) *url.URL {
+	u, err := url.Parse(path)
+	// Something is busted with the path, but it's too late to reject it. Pass it along as is.
+	//
+	// This construction of a URL may be wrong in some cases, but it preserves
+	// legacy prober behavior.
+	if err != nil {
+		u = &url.URL{
+			Path: path,
+		}
```

Comment:
can we just return the err and do not proceed with the wrong path? I think we can add more strict checking in Kruise, since we don't have some legacy cases to support as in kubernetes code

Reply:
done